### PR TITLE
Changes made to support WatchKit for different environments. Also cleane...

### PR DESCRIPTION
**Pull request by itschaitanya**
Adding support for WatchKit and cleaned up code to compile app extensions.

### CODE REVIEW

**Comment by icecrystal23** [comment]
This is our 2nd attempt at #120 with the addition of supporting changing entitlements to support WatchKit develop in a mixed enterprise / non-enterprise environment.

File: plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy [hunk 1]
```diff
@@ -70,11 +70,17 @@ class XcodeProjectFile {
 					project.xcodebuild.productName = config.getString("objects." + target + ".productName")
 				}
 				String type = config.getString("objects." + target + ".productType")
-				if (type.equalsIgnoreCase("com.apple.product-type.app-extension")) {
-					project.xcodebuild.productType = "appex"
-				}
+//				if (type.equalsIgnoreCase("com.apple.product-type.app-extension")) {
+//					project.xcodebuild.productType = "appex"
+//				}
+
```
Comment (renep) [inline plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy]: Why is this code a problem had you have disabled it?

File: plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy [hunk 2]
```diff
@@ -70,11 +70,17 @@ class XcodeProjectFile {
...
+                if (project.xcodebuild.doesAppExtensionNeedConfiguration(targetName)) {
```
Comment (renep) [inline plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy]: I do not find the definition to this method.

**Comment by renep** [comment]
Some minor issues:

I use tabs in my code instead of spaces ;-)

Second I would be nice to have an example to see how this works. I have not had the time to check out WatchKit. Maybe you can modify the Example in iOS/Example or create a own simple example there, that would be great.

There is a parameter documentation in the Documentation folder, where the AppExtension parameters should be added.

File: plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy (hunk 1 above)
Comment (itschaitanya) [inline plugin/src/main/groovy/org/openbakery/XcodeProjectFile.groovy]: When we build the main iOS application, the extensions/widgets are targets that are built as well. The code iterates over all the available targets and if it finds an extension, it changes the productType to "appex" which should not be true as the main target still is of type "app".

**Comment by icecrystal23** [comment]
We should close this pull request; there is a cleaned up version now (take 3! lol)

**Pull request closed without merging**